hardhat-vultisig/contracts/Whitelist.sol

```diff
@@ -11,24 +11,21 @@ import {IOracle} from "./interfaces/IOracle.sol";
  * - Ownable: Add whitelisted/blacklisted addresses
  * - Ownable: Set max ETH amount to buy(default 3 ETH)
  * - Ownable: Set univ3 TWAP oracle
- * - Vultisig contract `_beforeTokenTransfer` hook will call `checkWhitelist` function and this function will check if buyer is eligible
+ * - Token contract `_beforeTokenTransfer` hook will call `checkWhitelist` function and this function will check if buyer is eligible
  */
 contract Whitelist is Ownable {
     error NotWhitelisted();
     error Locked();
-    error NotVultisig();
-    error SelfWhitelistDisabled();
+    error NotToken();
     error Blacklisted();
     error MaxAddressCapOverflow();
 
     /// @notice Maximum ETH amount to contribute
     uint256 private _maxAddressCap;
     /// @notice Flag for locked period
     bool private _locked;
-    /// @notice Flag for self whitelist period
-    bool private _isSelfWhitelistDisabled;
```

Comment:
Self-whitelist and backlist are not needed.
We only need a list of whitelists.

Btw just take a look at Oracle contract, the functions seems quite simple. 
If we could combine it here, it's simpler for the deployments